### PR TITLE
Refactor to nwjs phoenix builder

### PR DESCRIPTION
**Pull request by insightfuls**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,6 +15,46 @@ var NodeWebkitBrowser = function(baseBrowserDecorator, args,config) {
...
+      useCaches: true,
+      showProgress: true
+    }))
+    .fetchAndExtract()
+    .then(function(extracted) {
+      return builder.findExecutable(process.platform, extracted);
+    })
+    .then(function(found) {
+      callback(null, found);
+    }, function(err) {
```
Comment (kierans) [inline index.js]: IMO using the `catch` ability from [Promises](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise) is more readable than the second arg to `then`

File: index.js [hunk 2]
```diff
@@ -65,15 +108,15 @@ NodeWebkitBrowser.prototype = {
   name: 'NW.js',
 
   DEFAULT_CMD: {
-    linux: findpath(),
-    darwin: findpath(),
-    win32: findpath()
+    linux: null,
```
Comment (kierans) [inline index.js:110]: Would it be better to have an empty object (with appropriate docs explaining what's going on)?

File: index.js (hunk 1 above)
Comment (insightfuls) [reply index.js]: Actually, that catches a bit more in this instance, so is the correct thing to do. Sometimes it isn't, so I have a habit of using the second arg.

File: index.js (hunk 2 above)
Comment (insightfuls) [reply index.js:110]: I presume this is conforming to a contract with the superclass which `_getCommand` uses, so I prefer to leave it. I think `null` is a better indicator than `undefined`, as it demonstrates it has deliberately fallen back to this default, whereas `undefined` could result from numerous additional unintended scenarios.

Comment (kierans) [reply index.js:110]: Should we add any docs then?

Comment (insightfuls) [reply index.js:110]: Not sure what I'd put that isn't either a bit of a guess, or obvious from a quick inspection of the rest of the code in the file. IMHO it's OK as it is.

**Pull request merged**